### PR TITLE
Got 200/200

### PR DESCRIPTION
**Pull request by kuv2707**
I have passed all the tests except one.

### CODE REVIEW

**Comment by EshaanAgg** [comment]
@kuv2707 Great going! Please pull the updates from the latest commits into your repository and then update this PR to reflect the same. Any commits that you push to your main branch will be updated automatically in this PR. 

Try to get them all. Best of luck!

File: src/strangerThings.js [hunk 1]
```diff
@@ -11,7 +11,10 @@ function sleep(ms) {
 TODO:
 This function should return a list of all the distinct characters in UPPERCASE that have been typed in the textbox with the id "message"
 */
-function getCharacters() {}
+function getCharacters()
+{
+	return document.querySelector("#message").value.split("");
+}
```
Comment (EshaanAgg) [inline src/strangerThings.js:17]: The characters must be converted to uppercase too.

File: src/strangerThings.js [hunk 2]
```diff
@@ -53,8 +56,15 @@ async function setCSS(character) {
 
 async function illuminateLight(index) {
 	const message = getCharacters();
-
+	console.log(message)
```
Comment (EshaanAgg) [inline src/strangerThings.js:59]: Remove `console.log`s before making a PR.

File: src/tempCodeRunnerFile.js [hunk 3]
```diff
@@ -0,0 +1 @@
+console
```
Comment (EshaanAgg) [inline src/tempCodeRunnerFile.js:1]: Delete this file as well.

File: src/array.js [hunk 4]
```diff
@@ -59,7 +63,7 @@ function getArrayOfPositives(arr) {
  *    [ false, 0, NaN, '', undefined ]   => [ ]
  */
 function removeFalsyValues(arr) {
-	throw new Error("Not implemented");
+	return arr.filter(elem=> (elem!=null && elem!=undefined && elem!=false && elem!='' && (""+elem)!="NaN" && elem!=0))
```
Comment (EshaanAgg) [inline src/array.js:66]: You don't need to check soo many conditions. JS treats all these values as `falsy`, i.e. equivalent to false. Try to read about it online.

File: src/array.js [hunk 5]
```diff
@@ -42,7 +46,7 @@ function doubleArray(arr) {
  *    [] => []
  */
 function getArrayOfPositives(arr) {
-	throw new Error("Not implemented");
+	return arr.filter((elem)=>elem>0)
```
Comment (EshaanAgg) [inline src/array.js:49]: Great use of iterator. Try to use them as much as possible!

File: src/array.js [hunk 6]
```diff
@@ -89,7 +93,7 @@ function getStringsLength(arr) {
  *   [ 1, 10, 100, 1000 ]  => 1111
  */
 function getItemsSum(arr) {
-	throw new Error("Not implemented");
+	return arr.reduce((val,sum)=>sum+val,0)
```
Comment (EshaanAgg) [inline src/array.js:96]: Great job.

File: src/conditionalAndLoops.js [hunk 7]
```diff
@@ -110,7 +131,30 @@ function reverseString(str) {
...
+		else
+		{
+			let k=stack.pop();
+			let clo=closerOf(k);
+			if(c!=clo)
+			return false
+		}
+	}
+	return(stack.length==0)
+
```
Comment (EshaanAgg) [inline src/conditionalAndLoops.js:157]: Good job. Kudos!

File: src/conditionalAndLoops.js [hunk 8]
```diff
@@ -169,7 +243,17 @@ function timespanToHumanString(startDate, endDate) {
  *    365, 10 => '365'
  */
 function toNaryString(num, n) {
-	throw new Error("Not implemented");
+	return ""+DecToBASE(num,n)
```
Comment (EshaanAgg) [inline src/conditionalAndLoops.js:246]: You can type cast to string in other ways too!
A bonus challenge: Will your implementation work for hexadecimal (base 16)?
Try to update your implementation to support all bases till 16.

File: src/numbers.js [hunk 9]
```diff
@@ -26,7 +26,12 @@ function getRectangleArea(width, height) {
  *  -3, 3  => 0
  */
 function getAverage(value1, value2) {
-	throw new Error("Not implemented");
+	if(value1+value2>Number.MAX_VALUE)
+	{
+		return Number.MAX_VALUE+((value1-Number.MAX_VALUE)+(value2-Number.MAX_VALUE))/2
+	}
+	else
+	return (value1+value2)/2;
```
Comment (EshaanAgg) [inline src/numbers.js:34]: Great job catching the overflow.

**Review by EshaanAgg — COMMENTED** [review]
Loved your attempt to solve all the problems. Left some reviews as to how you can start to write more idiomatic code for the long run! It was fun reviewing your code! Kudos.

File: src/strangerThings.js (hunk 1 above)
Comment (kuv2707) [reply src/strangerThings.js:17]: Ohh okay
I would just add .map(k=>k.toUpperCase());

File: src/conditionalAndLoops.js (hunk 8 above)
Comment (kuv2707) [reply src/conditionalAndLoops.js:246]: Done.
I made a function which intercepts the remainder and maps [10...15] to [A...F]

Comment (EshaanAgg) [reply src/conditionalAndLoops.js:241]: Great job.

File: src/strangerThings.js (hunk 1 above)
Comment (EshaanAgg) [reply src/strangerThings.js]: Great.

**Comment by kuv2707** [comment]
I scored 200/200 now.
I have also implemented the changes suggested by you

**Pull request closed without merging**